### DISCUSSION TITLE
What Will We "Do" with V2 Accounts as We Move to V3?

### DISCUSSION THREAD

**Discussion by BethanyG** [codebuddies/backend#152] [Q&A]
From closed issue #103 

![image](https://user-images.githubusercontent.com/5923094/90346480-17029380-dfde-11ea-993e-dad5e0764674.png)

**Comment by billglover** [comment]
Before we work on technical "solutions" to the user account question, it would be good to understand if we want to bring accounts over at all. Are we migrating any content from V2 to V3?

**Reply by BethanyG** [reply]
Oh -- hard agree on that question @billglover !  We may want to "start fresh" -- especially because of the issues we've had in the past with duplicate and weird-state users.  

We'll also have an email verification process for V3 - and new "terms and conditions" and GDPR disclosures, opt-ins etc.  So it may not be the greatest to "port" accounts, but rather use the info to contact community members and ask them to sign up for the new V3.    

@lpatmo - thoughts on that?

**Reply by lpatmo** [reply]
I think it would be an interesting technical challenge to port some things (I'm thinking user accounts, learnings, and hangout titles) to the new site, but I don't think it's a "business" priority -- I agree with @BethanyG that we can simply email community members to ask them to sign up to the new V3, and hopefully let the new features on the new site stand for itself.

**Reply by lpatmo** [reply]
(Also, thank you for moving this over to Discussions / restarting this discussion)

**Reply by BethanyG** [reply]
> I think it would be an interesting technical challenge to port some things (I'm thinking user accounts, learnings, and hangout titles)

Agree....at least on the Hangout titles and learnings.  

BUUUT...IIRC, we used social auth for V2 user accounts - based on Slack IDs.  And I don't think that always captured an email -- did it??  So - unless we want to write code to do that Slack ID lookup and find the associated username and email (since both will be required for V3), then make sure that those users are flagged on login so that they agree to the terms/conditions/etc and verify their email .. well, it goes from interesting technical challenge to not at all interesting administrative slog.  😁     We might just want to bite the bullet and send out communication based on the emails we have in Slack - introducing the shiny new website and all its cool features...and a link to get them started signing up.

**Reply by lpatmo** [reply]
Both Slack and GitHub auth did capture an email! :) I posted below an example of a user document, although the formatting got messed up :/ 

There are still more Slack emails than codebuddies.org accounts though, so at the end of the day it doesn't matter all that much. Growth trajectory: very slow. 😂 

V3: hopefully better engagement

**Comment by lpatmo** [comment]
PS if anyone's curious, we have 6288 user documents in the production Mongo database, and an example user document looks like: 

```
_id
:
"ARnxFh4bHwrLP"
createdAt
:
2017-08-25T19:32:28.240+00:00
services
:
Object
slack
:
Object
id
:
"U6BUXFK0R"
accessToken
:
"x-1882919-dfdec6b59059a4ade406358fa3..."
resume
:
Object
loginTokens
:
Array
roles
:
Object
CB
:
Array
username
:
"linda"
profile
:
Object
email
:
"example.email@gmail.com"
status
:
Object
online
:
false
lastLogin
:
Object
```

**Comment by angelocordon** [comment]
I am open to having a fresh slate for v3 (as stated [here](https://github.com/codebuddies/backend/discussions/152#discussioncomment-53567)). If so, we should probably think about whether there's any kind of data that folks would want to be exported out of v2? Is there a scenario here that we wouldn't need to at all? 🤞🏽 It would be great to just start completely fresh but not sure how responsible that would be.

**Reply by lpatmo** [reply]
> whether there's any kind of data that folks would want to be exported out of v2?

Some things I'm expecting to be on the todo list:
1. do a data dump (for sure)
2. reach out to active recent hangout creators and ask if there is any data they'd like to be saved
3. Send an email to everyone about CBV3

fwiw my guess is _probably_ there won't be much data that'll be VITAL to export to V3, but would be a courtesy to let active users (the hangout creators) know, and offer to share that data with them.

**Reply by angelocordon** [reply]
Agree, I don't think we need things to go into v3 from v2 -- was mainly thinking that we had a "notes feature" and maybe some folks would want them. 

Alternatively, can we host it for X-number of months in a different URL (v2.codebuddies.org?) so that people can still access it while v3 is up and running and they can get what they need out of it whatever way they can?
